python/cudf/cudf/core/frame.py

```diff
@@ -2280,19 +2321,42 @@ def drop_duplicates(
         if len(subset_cols) == 0:
             return self.copy(deep=True)
 
-        result = self.__class__._from_data(
-            *libcudf.stream_compaction.drop_duplicates(
-                self,
-                keys=subset,
-                keep=keep,
-                nulls_are_equal=nulls_are_equal,
-                ignore_index=ignore_index,
-            )
+        keys = self._positions_from_column_names(
+            subset, include_index=not ignore_index
+        )
+        result = self._drop_duplicates(
+            keys=keys,
+            keep=keep,
+            nulls_are_equal=nulls_are_equal,
+            ignore_index=ignore_index,
         )
-
         result._copy_type_metadata(self)
+        if self._index is not None:
+            result._index.name = self._index.name
+            if isinstance(self._index, cudf.MultiIndex):
+                result._index.names = self._index.names
         return result
 
+    def _positions_from_column_names(self, column_names, include_index=False):
+        """Map each column name into their positions in the frame.
+        """
```

Comment:
Only wrap docstrings' ending quotes if they exceed one line. Newer versions of `black` enforce this automatically but we haven't upgraded yet.
```suggestion
        """Map each column name into their positions in the frame."""
```